packages/dapp/src/stores/menu/MenuStore.tsx

```diff
@@ -22,88 +23,91 @@ export const defaultMenu: DefaultMenuType = {
   anchorRef: null,
 };
 
-export const useMenuStore = create<MenuState>((set, get) => ({
-  ...defaultMenu,
-  setValue: (key, value) =>
-    set(() => ({
-      [key]: value,
-    })),
-  setValues: (values) =>
-    set((state) => {
-      const updatedState: MenuProps = { ...state };
-      for (const key in values) {
-        if (Object.hasOwn(state, key)) {
-          updatedState[key] = values[key];
+export const useMenuStore = createWithEqualityFn<MenuState>(
```

Comment:
To have a previous behavior it is suggested to change it to `createWithEqualityFn(..., Object.is)`. If we make it `createWithEqualityFn(..., shallow)` then every usage of this hook will use `shallow` for comparison and I'm not sure we want it that way. Do we? 🤔

Reply:
Well, okay.. This was the reason I replaced those shallow´s on all those items. I definitely left it out for our Zustand store on "activeTabs" which really doesn´t need to be set to "shallow". But what´s with the others? Let´s discuss and see which works better